### PR TITLE
Feature/CTRL-90 - allow etcd image and version to be specified

### PR DESCRIPTION
**Pull request by Mojachieee**

### CODE REVIEW

File: cmd/plugin/cli/install.go [hunk 1]
```diff
@@ -209,13 +212,21 @@ func setInstallValues(cmd *cobra.Command, config *apiv1.KubectlStorageOSConfig)
...
 		config.Spec.Install.AdminUsername = cmd.Flags().Lookup(installer.AdminUsernameFlag).Value.String()
 		config.Spec.Install.AdminPassword = cmd.Flags().Lookup(installer.AdminPasswordFlag).Value.String()
 		config.Spec.Install.PortalClientID = cmd.Flags().Lookup(installer.PortalClientIDFlag).Value.String()
 		config.Spec.Install.PortalSecret = cmd.Flags().Lookup(installer.PortalSecretFlag).Value.String()
 		config.Spec.Install.PortalTenantID = cmd.Flags().Lookup(installer.PortalTenantIDFlag).Value.String()
 		config.Spec.Install.PortalAPIURL = cmd.Flags().Lookup(installer.PortalAPIURLFlag).Value.String()
 		config.InstallerMeta.StorageOSSecretYaml = ""
+
+		config.Spec.Install.EtcdVersionTag = cmd.Flags().Lookup(installer.EtcdVersionTag).Value.String()
+		// Perform the same validation as the etcd operator does, to ensure the install we succeed
```
Comment (Ricardo-Osorio) [inline cmd/plugin/cli/install.go:225]: typo? "the install we succeed"

File: api/v1/kubectlstorageosconfig_types.go [hunk 2]
```diff
@@ -80,6 +80,8 @@ type Install struct {
 	EtcdEndpoints                   string `json:"etcdEndpoints,omitempty"`
 	EtcdTLSEnabled                  bool   `json:"etcdTLSEnabled,omitempty"`
 	EtcdSecretName                  string `json:"etcdSecretName,omitempty"`
+	EtcdDockerRepository            string `json:"etcdDockerRepository,omitempty"`
```
Comment (Ricardo-Osorio) [inline api/v1/kubectlstorageosconfig_types.go:83]: Small detail
In `storageos/node:latest` the repository is "storageos", shouldn't this be named "image" instead?

**Review by Ricardo-Osorio — APPROVED** [review]

File: api/v1/kubectlstorageosconfig_types.go (hunk 2 above)
Comment (Mojachieee) [reply api/v1/kubectlstorageosconfig_types.go:83]: Yeah it's a bit ambiguous, but I used the same terminology the etcd-operator uses (the flag it uses is `--etcd-repository`).

 I'm happy to change it if other agree that it's confusing

Comment (nolancon) [reply api/v1/kubectlstorageosconfig_types.go:83]: I would have considered `storageos` to be the organization, and `node` to be the repo.... but maybe my whole life has been a lie 🤯

Comment (nolancon) [reply api/v1/kubectlstorageosconfig_types.go:83]: Also, do we need to specify the repository? surely the tag is enough?

Comment (Mojachieee) [reply api/v1/kubectlstorageosconfig_types.go:83]: Yeah this is a requirement from customer success. Some customers like to host their own images

**Review by nolancon — APPROVED** [review]

**Pull request merged**